### PR TITLE
Api and sdk updates for bicep client

### PR DESCRIPTION
**Pull request by anamikapan**
Requested changes are related to swagger - https://github.com/Azure/azure-rest-api-specs/pull/28204

### CODE REVIEW

**Review by jsquire — DISMISSED** [review]
@anamikapan :

Your account lacks the required public GitHub organizations and permissions required of an internal contributor.  Please review the Azure SDK onboarding documentation and use the associated Teams channel for support.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

You can verify the state of your account by running the [Validate-AzsdkCodeOwner](https://github.com/Azure/azure-sdk-tools/blob/main/tools/github/scripts/Validate-AzsdkCodeOwner.ps1) script from the Azure SDK tools repository.

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

**Comment by anamikapan** [comment]
@jsquire I have requested the specific permissions. Also, the build failures on the PR are because of some spell checks and words that are a part of swagger changes. I have no control over that. Can we manually try to override the spell check errors ?

**Comment by jsquire** [comment]
@anamikapan :  You now have the correct permissions; however you still have not made your Azure and Microsoft org memberships public.   Please use the script that I referenced above to validate your account after updating your settings.

**Comment by jsquire** [comment]
> Can we manually try to override the spell check errors ?

You'll want to update the [cspell config](https://github.com/Azure/azure-sdk-for-net/blob/main/.vscode/cspell.json) with exceptions specific to this library.   Please follow the existing pattern in the file.

**Comment by anamikapan** [comment]
@jsquire should cspell config additions be part of the same PR? or should that go in before this. can you please clarify

**Comment by ArthurMa1978** [comment]
> @anamikapan : You now have the correct permissions; however you still have not made your Azure and Microsoft org memberships public. Please use the script that I referenced above to validate your account after updating your settings.
> 
> @ArthurMa1978 : Can you confirm whether or not your team will own this library going forward?

No, the service team is responsible for managing the library, and we will assist with the release process.

**Comment by jsquire** [comment]
> > @anamikapan : You now have the correct permissions; however you still have not made your Azure and Microsoft org memberships public. Please use the script that I referenced above to validate your account after updating your settings.
> > @ArthurMa1978 : Can you confirm whether or not your team will own this library going forward?
> 
> No, the service team is responsible for managing the library, and we will assist with the release process.

Thanks, Arthur.   

@anamikapan:  Arthur's response indicates that you will also be required to provide CODEOWNERS data for this PR to move forward.

**Comment by jsquire** [comment]
> @jsquire should cspell config additions be part of the same PR? or should that go in before this. can you please clarify

Please include them as part of this PR.

**Comment by majastrz** [comment]
@jsquire I'm one of the engineering managers for Azure Deployments. The ARM package (sdk/resources/Azure.ResourceManager.Resources/**) that is modified by @anamikapan in this PR is owned by multiple teams within the ARM organization. I'm perfectly happy to take ownership for the parts of the package that my team owns and add ourselves to the list but we cannot be responsible for the whole package.

The parts that we can own are the paths related to the following resource types (and any nested resource types below them) in the `Microsoft.Resources` namespace:
* `deployments`
* `templateSpecs`
* `builtInTemplateSpecs`
* `deploymentScripts`
* `deploymentStacks`
* `decompileBicep`

**Comment by jsquire** [comment]
> @jsquire I'm one of the engineering managers for Azure Deployments. The ARM package (sdk/resources/Azure.ResourceManager.Resources/**) that is modified by @anamikapan in this PR is owned by multiple teams within the ARM organization. I'm perfectly happy to take ownership for the parts of the package that my team owns and add ourselves to the list but we cannot be responsible for the whole package.
> 
> The parts that we can own are the paths related to the following resource types (and any nested resource types below them) in the `Microsoft.Resources` namespace:
> 
> * `deployments`
> * `templateSpecs`
> * `builtInTemplateSpecs`
> * `deploymentScripts`
> * `deploymentStacks`
> * `decompileBicep`

Thanks, @majastrz.   There will need to be a CODEOWNERS entry added as part of this PR around L965 in the management package section which follows the existing format.    Since a library is represented by a single label for issue support purposes, I'd suggest that you create a GitHub team and use that as the owner so that you can more easily manage adding/removing contacts from it.

**Comment by majastrz** [comment]
> Thanks, @majastrz. There will need to be a CODEOWNERS entry added as part of this PR around L965 in the management package section which follows the existing format. Since a library is represented by a single label for issue support purposes, I'd suggest that you create a GitHub team and use that as the owner so that you can more easily manage adding/removing contacts from it.

@jsquire We have an existing team (@Azure/bicep-admins) that covers Azure Deployments functionality. Can we specify multiple teams?

**Comment by jsquire** [comment]
> > Thanks, @majastrz. There will need to be a CODEOWNERS entry added as part of this PR around L965 in the management package section which follows the existing format. Since a library is represented by a single label for issue support purposes, I'd suggest that you create a GitHub team and use that as the owner so that you can more easily manage adding/removing contacts from it.
> 
> @jsquire We have an existing team (@Azure/bicep-admins) that covers Azure Deployments functionality. Can we specify multiple teams?

 I believe so.   @JimSuplizio  - can you confirm?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -913,7 +913,7 @@
 /**/*Management*/                                                  @archerzz @ArcturusZhang @ArthurMa1978
 
 # PRLabel: %Mgmt
-/**/Azure.ResourceManager*/                                        @archerzz @ArcturusZhang @ArthurMa1978
+/**/Azure.ResourceManager*/                                        @archerzz @ArcturusZhang @ArthurMa1978 @bicep-admins
```
Comment (jsquire) [inline .github/CODEOWNERS:916]: You do not want to do this.   That will include your team for ALL resource manager libraries in the repository.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -913,7 +913,7 @@
...
 
 # -----------------------------------------------------------------
```
Comment (jsquire) [inline .github/CODEOWNERS:918]: Please add a dedicated section for this specific path and label below L927, preserving alphabetical ordering by tag.

Comment (shenglol) [reply .github/CODEOWNERS:918]: @anamikapan We need to add `@Azure/bicep-admins` as an owner for the `/sdk/resources/Azure.ResourceManager.Resources/` path. Please ensure that the team name is fully qualified to include the `Azure` organization.

Comment (majastrz) [reply .github/CODEOWNERS:918]: @anamikapan Let's use @azure/arm-sdk-owners instead. I just created it.

File: sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/ResourcesExtensions.cs [hunk 3]
```diff
@@ -1231,6 +1231,70 @@ public static Pageable<JitRequestResource> GetJitRequestDefinitions(this Subscri
...
+        /// <item>
+        /// <term>Mocking</term>
+        /// <description>To mock this method, please mock <see cref="MockableResourcesSubscriptionResource.BicepDecompile(DecompileOperationContent,CancellationToken)"/> instead.</description>
+        /// </item>
+        /// </summary>
+        /// <param name="subscriptionResource"> The <see cref="SubscriptionResource" /> instance the method will execute against. </param>
+        /// <param name="content"> Decompile operation request supplied to the operation. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="subscriptionResource"/> or <paramref name="content"/> is null. </exception>
+        public static async Task<Response<DecompileOperationSuccessResult>> BicepDecompileAsync(this SubscriptionResource subscriptionResource, DecompileOperationContent content, CancellationToken cancellationToken = default)
```
Comment (ArthurMa1978) [inline sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/ResourcesExtensions.cs:1259]: Please add tests for this new operation

**Comment by JimSuplizio** [comment]
> Since a library is represented by a single label for issue support purposes, I'd suggest that you create a GitHub team and use that as the owner so that you can more easily manage adding/removing contacts from it.

So long as the team is under the azure-sdk-write team it'll get expanded into the individual users.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -966,6 +966,15 @@
...
 
+# PRLabel: %ARM
+/sdk/resources/Azure.ResourceManager.*/                            @anamikapan
+
+# ServiceLabel: %ARM %Mgmt
+# ServiceOwners:                                                   @anamikapan
+
+# PRLabel: %Recovery Services Backup
+/sdk/recoveryservices-backup/Azure.ResourceManager.*/              @Daya-Patil
+
```
Comment (jsquire) [inline .github/CODEOWNERS:977]:
```suggestion
```

This library is not related to your changes.

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -966,6 +966,15 @@
 # ServiceOwners:                                                   @jamesfan1 @bangbingsyb @spencerofwiti
```
Comment (jsquire) [inline .github/CODEOWNERS:968]:
```suggestion
```

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -966,6 +966,15 @@
...
 
 
+# PRLabel: %ARM
```
Comment (jsquire) [inline .github/CODEOWNERS:969]: Please move this to L929 to preserve alphabetical ordering by label.

File: .github/CODEOWNERS (hunk 4 above)
Comment (anamikapan) [reply .github/CODEOWNERS:977]: @jsquire arthur suggested this change for some reason as it was showing some errors in codeowners file and carry forward those changes. I can remove it

Comment (jsquire) [reply .github/CODEOWNERS:977]: Adding a PR entry where one didn't exist won't resolve a linter error.   If we're seeing errors pop up, it likely means that someone left their team or the company and we've got to scrub data.   I don't see an error to fix at the moment.

Comment (jsquire) [reply .github/CODEOWNERS:977]: That said, if Arthur says this person should be on the hook for reviews, then I'm good leaving it.

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/ResourcesExtensions.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/ResourcesExtensions.cs:1259]: Will add the test in another pr
